lib/url.c

```diff
@@ -455,8 +455,8 @@ CURLcode Curl_close(struct Curl_easy **datap)
   }
 #endif
 
+#ifndef CURL_DISABLE_FORM_API
```

Comment:
Should this not rather check `CURL_DISABLE_MIME` ?

Reply:
At least with `./configure` if you try to build with the form API enabled but with MIME disabled, it fails with:

```
configure: error: MIME support needs to be enabled in order to enable form API support
```

So I don't think there is an issue in this case.

Reply:
Ah yes, because the `formp` pointer is for the form API only.

However, this breaks the **linux ubuntu disable_http_smtp_imap) ** build because you removed the `CURL_DISABLE_HTTP` check.